In [1]:
import matplotlib.pyplot as plt
import sys

sys.path.append('../../')
from busi.air.air import Air
from busi.air.temperature import Temperature

# Manual

## Set

At first, set the parameters of constructor.  
The first parameter is temperature as 'Temperature' class.  
The second parameter is humidity. Which this value means relative (%) or absolute (kg/kgDA) is depending on the therd parameter.  
The therd parameter represents the second parameter's meaning. You can specify 'relative' or 'absolute'.

The below is an example.

In [2]:
a = Air(t=Temperature(35.0, 'Celsius'), h=50.0, humtype='relative')

Relative humidity shall be 0 ~ 100 (%).

In [3]:
try:
    Air(Temperature(35.0, 'Celsius'), -1.0, 'relative')
except ValueError as e:
    print(e)

Error: Relative humidity shall be equal to or more than 0 %.


In [4]:
try:
    Air(Temperature(35.0,'Celsius'), 101.0, 'relative')
except ValueError as e:
    print(e)

Error: Relative humidity shall not be over 100 %.


Absolute humidity shall not be over the saturated absolute humidity. 

In [5]:
try:
    Air(Temperature(35.0, 'Celsius'), 0.037, 'absolute')
        # saturated absolute humidity at 35 degree C is 0.036588480692732184 based on SONNTAG equation.
except ValueError as e:
    print(e)

Error: Absolute humidity shall be not set at the state that the relative humidity is over 100 %.


## Get

You can get the parameters by using property.

### Temperature

In [6]:
a.T, a.T.C, a.T.K  # class, degree C, K 

(<busi.air.temperature.Temperature at 0x1b041920bb0>, 35.0, 308.15)

### Relative Humidity

In [7]:
a.RH # %

50.0

### Absolute Humidity

In [8]:
a.AH # kg/kgDA

0.017771544359487364

### Vapor Pressure

In [9]:
a.VP # Pa

2814.601162088015

### Saturated Vapor Pressure

In [10]:
a.SVP # Pa

5629.20232417603

### Saturated Absolute Humidity

In [11]:
a.SAH # kg/kgDA

0.036588480692732184

### Dew Point Temperature

In [12]:
a.DPT, a.DPT.K, a.DPT.C

(<busi.air.temperature.Temperature at 0x1b044456f40>,
 296.17100593817213,
 23.021005938172152)

## Change Parameter

You can change parameters.

### Temperature

In the example below, the temperature is changed to 40.0 degree C from 35.0 degree C. 

In [13]:
a2 = Air(Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [14]:
a2.T.C, a2.RH, a2.AH

(35.0, 50.0, 0.017771544359487364)

In [15]:
a2.T = Temperature(40.0,'Celsius')

In [16]:
a2.T.C, a2.RH, a2.AH

(40.0, 38.110878995581295, 0.017771544359487364)

You can not set the temperature which is below the dew point temperature.

In [17]:
a3 = Air(Temperature(35.0,'Celsius'), 50.0, 'relative')

In [18]:
a3.DPT.C

23.021005938172152

In [19]:
try:
    a3.T = Temperature(23.0,'Celsius') # The dew point temperature is 23.02101573944094 degree C.
except ValueError as e:
    print(e)

Error: Temperature shall be not set at the state that the relative humidity is over 100 %.


### Absolute Humidity

In the example below, the absolute humidity is changed to 0.019 kg/kgDA from 0.017 kg/kgDA. 

In [20]:
a4 = Air(Temperature(35.0,'Celsius'), 0.017, 'absolute')

In [21]:
a4.T.C, a4.RH, a4.AH

(35.0, 47.88702060713158, 0.017)

In [22]:
a4.AH = 0.019

In [23]:
a4.T.C, a4.RH, a4.AH

(35.0, 53.353796199980216, 0.019)

You can not set the absolute humidity which is over the saturated absolute humidity.

In [24]:
a5 = Air( Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [25]:
a5.SAH

0.036588480692732184

In [26]:
try:
    a5.AH = 0.0366 # The saturated absolute humidity is 0.036588480692732184(kg/kgDA).
except ValueError as e:
    print(e)

Error: Absolute humidity shall be not set at the state that the relative humidity is over 100 %.


### Relative Humidity

In the example below, the relative humidity is changed to 75.0 % from 50.0 %. 

In [27]:
a6 = Air( Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [28]:
a6.T.C, a6.RH, a6.AH

(35.0, 50.0, 0.017771544359487364)

In [29]:
a6.RH = 75.0

In [30]:
a6.RH

75.0

You can not set the relative humidity which is out of range of 0 ~ 100 %.

In [31]:
a7 = Air( Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [32]:
try:
    a7.RH = -1.0
except ValueError as e:
    print(e)

Error: Relative humidity shall be equal to or more than 0 %.


In [33]:
try:
    a7.RH = 101.0
except ValueError as e:
    print(e)

Error: Relative humidity shall not be over 100 %.


### Vapor Pressure

In the example below, the vapor pressure is changed to 3000.0 Pa from 2814.601162088015 Pa. 

In [34]:
a8 = Air( Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [35]:
a8.VP

2814.601162088015

In [36]:
a8.VP = 3000.0

In [37]:
a8.VP

3000.0

You can not set the vapor pressure which is over the saturated vapor pressure.

In [38]:
a9 = Air( Temperature(35.0,'Celsius'), 50.0, 'relative' )

In [39]:
a9.SVP

5629.20232417603

In [40]:
try:
    a9.VP = 5630.0 # The saturated vapor pressure is 5629.20232417603(Pa).
except ValueError as e:
    print(e)

Error: Vapor pressure shall be not set at the state that the relative humidity is over 100 %.
